train_segmentation.py

```diff
@@ -485,146 +491,165 @@ def main(params, config_path):
+    eval_batch_size = get_key_def('eval_batch_size', cfg['trainer'], default=batch_size)
+    num_epochs = get_key_def('max_epochs', cfg['trainer'])
+    model_name = get_key_def('model_name', cfg['model']).lower()
+    # TODO need to keep in parameters? see victor stuff
+    # BGR_to_RGB = get_key_def('BGR_to_RGB', params['global'], expected_type=bool)
+    BGR_to_RGB = False
 
     # OPTIONAL PARAMETERS
-    # basics
-    debug = get_key_def('debug_mode', params['global'], default=False, expected_type=bool)
-    task = get_key_def('task', params['global'], default='segmentation', expected_type=str)
-    if not task == 'segmentation':
-        raise ValueError(f"The task should be segmentation. The provided value is {task}")
-    dontcare_val = get_key_def("ignore_index", params["training"], default=-1, expected_type=int)
-    crop_size = get_key_def('target_size', params['training'], default=None, expected_type=int)
-    batch_metrics = get_key_def('batch_metrics', params['training'], default=None, expected_type=int)
-    meta_map = get_key_def("meta_map", params["global"], default=None)
+    debug = get_key_def('debug', cfg)
+    task = get_key_def('task_name',  cfg['task'], default='segmentation')
+    dontcare_val = get_key_def("ignore_index", cfg['dataset'], default=-1)
+    bucket_name = get_key_def('bucket_name', cfg['AWS'])
+    scale = get_key_def('scale_data', cfg['augmentation'], default=[0, 1])
+    batch_metrics = get_key_def('batch_metrics', cfg['training'], default=None)
+    meta_map = get_key_def("meta_map", cfg['training'], default=None)  # TODO what is that?
+    crop_size = get_key_def('target_size', cfg['training'], default=None)
+    # if error
     if meta_map and not Path(meta_map).is_file():
-        raise FileNotFoundError(f'Couldn\'t locate {meta_map}')
-    bucket_name = get_key_def('bucket_name', params['global'])  # AWS
-    scale = get_key_def('scale_data', params['global'], default=[0, 1], expected_type=List)
-
-    # model params
-    loss_fn = get_key_def('loss_fn', params['training'], default='CrossEntropy', expected_type=str)
-    class_weights = get_key_def('class_weights', params['training'], default=None, expected_type=Sequence)
+        raise logging.critical(FileNotFoundError(f'\nCouldn\'t locate {meta_map}'))
+    if task != 'segmentation':
+        raise logging.critical(ValueError(f"\nThe task should be segmentation. The provided value is {task}"))
+
+    # MODEL PARAMETERS
+    class_weights = get_key_def('class_weights', cfg['dataset'], default=None)
+    loss_fn = get_key_def('loss_fn', cfg['training'], default='CrossEntropy')
+    optimizer = get_key_def('optimizer_name', cfg['optimizer'], default='adam')  # TODO change something to call the function
+    pretrained = get_key_def('pretrained', cfg['model'], default=True)
+    train_state_dict_path = get_key_def('state_dict_path', cfg['general'], default=None)
+    dropout_prob = get_key_def('factor', cfg['scheduler']['params'], default=None)
```

Comment:
Do we want to keep default values in the get_key_def() function? The yaml config files already set a default value for all of these parameters, right?

Reply:
True, but that can change depending of the task, so i think we should keep it just in case but can be ignore in the future. We can add that point on the débate in the next meeting.